sim/src/generic_eps_hardware_model.cpp

```diff
@@ -433,6 +480,61 @@ namespace Nos3
         return valid;
     }
 
+    void Generic_epsHardwareModel::update_battery_values(void)
+    {
+        //sim_logger->debug("Generic_epsHardwareModel::update_battery_values");
+        boost::shared_ptr<Generic_epsDataPoint> data_point = boost::dynamic_pointer_cast<Generic_epsDataPoint>(_generic_eps_dp->get_data_point());
+        double svb_X = (data_point->get_sun_vector_x() > 0) ? data_point->get_sun_vector_x() : 0.0;
+        double svb_minusX = (data_point->get_sun_vector_x() < 0) ? (-1)*data_point->get_sun_vector_x() : 0.0;
+        double svb_Y = (data_point->get_sun_vector_y() > 0) ? data_point->get_sun_vector_y() : 0.0;
+        double svb_Z = (data_point->get_sun_vector_z() > 0) ? data_point->get_sun_vector_z() : 0.0;
+
+        sim_logger->debug("Generic_epsHardwareModel::update_battery_values:  X = %.3f; Y = %.3f; Z = %.3f;", svb_X, svb_Y, svb_Z);
+
+        /* Note: Assuming solar arrays on all +/- X, Y, and Z faces */
+        // The "cosine effect" is the most relevant part, affecting the power 
+        // received. I have no idea if it impacts the voltage or the current,
+        // but theoretically it should not matter - I can just multiply times
+        // the whole thing.
+        // Technically, when the angle to the sun is greater than 45 degrees
+        // there begins to be significant light reflected away, an effect which
+        // is not replicated here.
+
+        double p_out = 0;
+
+        for (int i = 1; i < 4; i++)
+        {
+            p_out = p_out + (_bus[i]._voltage/1000.0)*(_bus[i]._current/1000.0);
+        }
+        for (int i = 0; i < 8; i++)
+        {
+            int switchonoff = (_switch[i]._status != 0) ? 1 : 0;
+            p_out = p_out + (_switch[i]._voltage/1000.0)*(_switch[i]._current/1000.0)*switchonoff;
+
+        }
+        
+        double p_in = _power_per_panel*svb_X + _power_per_panel*svb_minusX + _power_per_panel*svb_Y + _power_per_panel*svb_Z;
+        double delta_p = (0.01 * (p_in - p_out));
```

Comment:
Is 0.01 for 1/100th of a second?  If so, maybe use the config parameter for sim microseconds per tick (/1000000) instead.

Reply:
That makes sense; the relevant change has been made.